twitterpuller/src/test/java/de/brandwatch/minianalytics/twitterpuller/twitter/TweetFetcherTest.java

```diff
@@ -0,0 +1,52 @@
+package de.brandwatch.minianalytics.twitterpuller.twitter;
+
+
+import de.brandwatch.minianalytics.twitterpuller.kafka.Producer;
+import de.brandwatch.minianalytics.twitterpuller.model.Resource;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import static org.junit.Assert.*;
+
+import static org.mockito.Mockito.*;
+
+import org.mockito.junit.MockitoJUnitRunner;
+import twitter4j.Status;
+import twitter4j.User;
+
+import java.util.Date;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class TweetFetcherTest {
+
+    @Mock
+    Producer producer;
+
+    @Test
+    public void testIfResourceObjectIsGeneratedCorrectly() {
+
+        Status status = mock(Status.class, RETURNS_DEEP_STUBS);
```

Comment:
`RETURNS_DEEP_STUBS` is not considered to be good, clean style:

```
This feature should rarely be required for regular clean code! Leave it for legacy code.
     * Mocking a mock to return a mock, to return a mock, (...), to return something meaningful
     * hints at violation of Law of Demeter or mocking a value object (a well known anti-pattern).
```

Instead, below, you should mock a `User` independently.

I have the feeling that this doesn't even exist in mockito 3, although I am not 100% sure. But you seem to use an older version (2.x.x), which you should fix as well.